spec/requests/api/services_spec.rb

```diff
@@ -195,4 +197,63 @@ def format_retirement_date(time)
       expect(svc2.retirement_warn).to eq(5)
     end
   end
+
+  describe "Service reconfiguration" do
+    let(:dialog1) { FactoryGirl.create(:dialog, :label => "Dialog1") }
+    let(:tab1)    { FactoryGirl.create(:dialog_tab, :label => "Tab1") }
+    let(:group1)  { FactoryGirl.create(:dialog_group, :label => "Group1") }
+    let(:text1)   { FactoryGirl.create(:dialog_field_text_box, :label => "TextBox1", :name => "text1") }
+    let(:st1)     { FactoryGirl.create(:service_template, :name => "template1") }
+    let(:ra1) do
+      FactoryGirl.create(:resource_action, :action => "Reconfigure", :dialog => dialog1,
+                         :ae_namespace => "namespace", :ae_class => "class", :ae_instance => "instance")
+    end
+
+    it "rejects requests without appropriate role" do
+      api_basic_authorize
+
+      run_post(services_url(100), gen_request(:reconfigure))
+
+      expect_request_forbidden
+    end
+
+    it "does not return reconfigure action for non-reconfigurable services" do
+      api_basic_authorize
+      update_user_role(@role, action_identifier(:services, :retire), action_identifier(:services, :reconfigure))
+
+      run_get services_url(svc1.id)
+
+      expect_request_success
+      expect_result_to_have_keys(%w(actions))
+      expect(action_names(@result)).to match_array(%w(retire))
+    end
+
+    it "returns reconfigure action for reconfigurable services" do
+      api_basic_authorize
+      update_user_role(@role, action_identifier(:services, :retire), action_identifier(:services, :reconfigure))
+
+      st1.resource_actions = [ra1]
+      svc1.service_template_id = st1.id
+      svc1.save
+
+      run_get services_url(svc1.id)
+
+      expect_request_success
+      expect_result_to_have_keys(%w(actions))
+      expect(action_names(@result)).to match_array(%w(retire reconfigure))
```

Comment:
WDYT about:

```ruby
expect(@result).to have_actions("retire", "reconfigure"))

# ...

def have_actions(*names)
  include(:actions => a_collection_containing_exactly(names.map { |name| a_hash_including(:name => name) }))
end
```

would remove the need for the extra expectation (`have_keys(%w(actions))`) and the duplication of checking for that in `action_names`